.NET Aspire 9.0 - GA Release content (#1867)
dotnet/docs-aspire: docs/fundamentals/aspire-sdk-templates.md

---
title: .NET Aspire templates
description: Learn how to install the .NET Aspire templates, and how to use them to create new apps.
ms.date: 11/04/2024
zone_pivot_groups: dev-environment
uid: dotnet/aspire/templates
---

# .NET Aspire templates

There are a number of .NET Aspire project templates available to you. You can use these templates to create full .NET Aspire solutions, or add individual projects to existing .NET Aspire solutions.

The .NET Aspire templates are available in the [📦 Aspire.ProjectTemplates](https://www.nuget.org/packages/Aspire.ProjectTemplates) NuGet package.

## Install the .NET Aspire templates

:::zone pivot="visual-studio"

> The .NET Aspire templates are installed automatically when you install Visual Studio 17.9 or later.

Reviewer: As you and I chatted offline, this is not actually true. You can use aspire with 17.9 and above, but it is not guaranteed to have the templates automatically installed. The only version that comes with the templates installed built-in (and note that this is only the case if you check the web workload) is 17.12 GA.

Reviewer: The other thing that we talked about, is that for users that want to still use Aspire 8.x, they will now have to make sure they check the checkbox in the installer so that the Aspire workload is installed. Not sure if that is something we want to have in docs or not, specially given that when we ship 9.0 8.x will be out of support, but just mentioning it in case you think we should.

Reviewer: Not sure if this is too verbose

```suggestion
To install the .NET Aspire templates in Visual Studio, you need to manually install them unless you are using Visual Studio 17.12 or later. For Visual Studio 17.9 to 17.11, follow these steps:

1. Open Visual Studio.
2. Go to **Tools** > **NuGet Package Manager** > **Package Manager Console**.
3. Run the following command to install the templates:

  ```powershell
  dotnet new install Aspire.ProjectTemplates
  `` ****(add an extra backtick here. I ommited it so suggestion wouldn't look cropped.)****

For Visual Studio 17.12 or later, the .NET Aspire templates are installed automatically.
```

:::zone-end
:::zone pivot="vscode,dotnet-cli"

To install these templates, use the [dotnet new install](/dotnet/core/tools/dotnet-new-install) command, passing in the `Aspire.ProjectTemplates` NuGet identifier.

```dotnetcli

> dotnet new install Aspire.ProjectTemplates

Reviewer: Might be interesting to mention in here that you may require to run `--force` in the case the user already has the Aspire Workload installed, as otherwise they will get an issue telling them that installing this templates would override the ones from the workload.

Author: Added

```

:::zone-end

## List the .NET Aspire templates

:::zone pivot="visual-studio"

> The .NET Aspire templates are installed automatically when you install Visual Studio 17.9 or later. To see what .NET Aspire templates are available, select **File** > **New** > **Project** in Visual Studio, and search for "Aspire" in the search bar (<kbd>Alt</kbd>+<kbd>S</kbd>). You'll see a list of available .NET Aspire project templates:

Reviewer: Let me know if you want more details or if you want me to suggest changes here that represent the ways to get the templates, and you can review.

Author: Yes, please... Please make suggestions here in the PR, so that I can review/edit and accept.

Reviewer: ```suggestion
The .NET Aspire templates are installed automatically when you install Visual Studio 17.12 or later. To see what .NET Aspire templates are available, select **File** > **New** > **Project** in Visual Studio, and search for "Aspire" in the search bar (<kbd>Alt</kbd>+<kbd>S</kbd>). You'll see a list of available .NET Aspire project templates:
```

> If you already have the .NET Aspire workload installed, you need to pass the `--force` flag to overwrite the existing templates. Feel free to uninstall the .NET Aspire workload.

:::zone-end

## List the .NET Aspire templates

:::zone pivot="visual-studio"

The .NET Aspire templates are installed automatically when you install Visual Studio 17.9 or later. To see what .NET Aspire templates are available, select **File** > **New** > **Project** in Visual Studio, and search for "Aspire" in the search bar (<kbd>Alt</kbd>+<kbd>S</kbd>). You'll see a list of available .NET Aspire project templates:

:::image type="content" source="media/vs-create-dotnet-aspire-proj.png" alt-text="Visual Studio: Create new project and search for 'Aspire'." lightbox="media/vs-create-dotnet-aspire-proj.png":::

:::zone-end
:::zone pivot="vscode"

To view the available templates in Visual Studio Code with the C# DevKit installed, select the **Create .NET Project** button when no folder is opened in the **Explorer** view:

:::image type="content" source="media/vscode-create-dotnet-proj.png" alt-text="Visual Studio Code: Create .NET Project button." lightbox="media/vscode-create-dotnet-proj.png":::

> Then, search for "Aspire 9" in the search bar to see the available .NET Aspire project templates:

Reviewer: We should double check, but I don't think this is accurate as now everything is in a single template, so I don't think "9" is on the name.

```suggestion
Then, search for "Aspire" in the search bar to see the available .NET Aspire project templates:
```

:::image type="content" source="media/vscode-create-dotnet-aspire-proj.png" alt-text="Visual Studio Code: Create new project and search for 'Aspire'." lightbox="media/vscode-create-dotnet-aspire-proj.png":::

:::zone-end
:::zone pivot="dotnet-cli"

To verify that the .NET Aspire templates are installed, use the [dotnet new list](/dotnet/core/tools/dotnet-new-list) command, passing in the `aspire` template name:

```dotnetcli
dotnet new list aspire
```

Your console output should look like the following:

[!INCLUDE [dotnet-new-list-aspire-output](includes/dotnet-new-list-aspire-output.md)]

:::zone-end

> [!NOTE]

> > If you still have .NET Aspire 8.0 templates installed (from the workload), you'll see both .NET Aspire 8.0 and .NET Aspire 9.0 templates in the list. They can be distinguished by the version number in the template name.

Reviewer: This is no longer true, so we should remove.
